source/extensions/quic_listeners/quiche/envoy_quic_server_session.cc

```diff
@@ -39,8 +34,18 @@ std::unique_ptr<quic::QuicCryptoServerStreamBase>
 EnvoyQuicServerSession::CreateQuicCryptoServerStream(
     const quic::QuicCryptoServerConfig* crypto_config,
     quic::QuicCompressedCertsCache* compressed_certs_cache) {
-  return quic::CreateCryptoServerStream(crypto_config, compressed_certs_cache, this,
-                                        stream_helper());
+  switch (connection()->version().handshake_protocol) {
+  case quic::PROTOCOL_QUIC_CRYPTO:
+    return std::make_unique<EnvoyQuicCryptoServerStream>(crypto_config, compressed_certs_cache,
+                                                         this, stream_helper());
+  case quic::PROTOCOL_TLS1_3:
+    return std::make_unique<EnvoyQuicTlsServerHandshaker>(this, *crypto_config);
+  case quic::PROTOCOL_UNSUPPORTED:
+    break;
+  }
+  QUIC_BUG << "Unknown handshake protocol: "
+           << static_cast<int>(connection()->version().handshake_protocol);
+  return nullptr;
```

Comment:
I think return a nullptr may crash the program, how about just return a EnvoyQuicCryptoServerStream and let the handshake fail?